docs_src/design/ucr/Microservice-Authentication.md

```diff
@@ -0,0 +1,132 @@
+
+  _PR.AC-1: Identities and credentials are issued, managed, verified,
+  revoked, and audited for authorized devices, users, and processes._
+
+
+### Target Users
+- Device Owner
+- Device User
+- Device Maintainer
+- Service Provider
+
+### Description
+
+Miroservice authentication provides the following benfits,
+which are potentially valuable to all of the listed target users:
+
+- Provides a defense against malware running on the device,
+  as currently there is no mechanism to ensure that only
+  authorized users or processes are allowed to invoke EdgeX services.
+
+- Provides greater auditability as to who initiated a particular
+  action on the device.
+
+- Depending on implementation, may provide a way to revoke
+  access that was previously granted,
+  or allow customers to tie in to enterprise
+  identity management systems.
+
+
+### Existing solutions
+
+Microservice authentication is currently implemented around two primary vectors:
+
+- Token-based authentication schemes.
+  Initiator sends an identifier along with a request to the receiver.
+  The identifier is cryptographically validated using a key trusted by
+  the receiver, or the receiver asks a trusted third party to verify the identifier.
+
+  A benefit of token-based authentication schemes is identity delegation,
+  whereby the identifier can be passed through a chain of calls to
+  preserve the identity of the original initiator.
+  The identifier can often be tunned through other protocols.
+
+  A drawback of token-based authentication is that due to man-in-the-middle
+  threats, token-based authentication over an unencrypted network is insecure.
+  Another drawback of token-based authentication is that it is unidirectional:
+  the receiver can authenticate the initiator, but not vice-versa.
+
+- Mutual-auth TLS.
+  Both the intitiator and the receiver particpate in a session-oriented message
+  exchange based on public-key cryptography, where each trusts the other's
+  digital signature authority.
+
+  A benefit of mutual-auth TLS is that it enforces proof-of-possession
+  (of a cryptographic key), thus preventing token-stealing attacks.
+
+  A drawback of mutual-auth TLS is that it requires direct 
+  service-to-service network connectivity: services cannot be proxied
+  without losing the identity of the original caller.
+  Another drawback is that mutual-auth TLS requires encryption
+  for all use cases, including same-host communication,
+  which results in multiple redundant layers of encryption
+  when lower layers of the network stack are also encrypted.
+
+  This most onerous drawback to mutual-auth TLS is overhead associated with
+  certificate and key rotation on both the initiator and receiver when
+  the cryptoperiods of their cryptographic materials expire. 
+
```

Comment:
Should OpenZiti be add to this?

Reply:
It depends on how OpenZiti is configured.  If you provide the TLS days, you are responsible.  But if the service mesh itself handles it, it is expected that the mesh also takes care of key rotation automatically.

Reply:
I am say should OpenZiti be add to list of known solutions.

Reply:
No, not generally.  Some service meshes just wrap the service they are protecting and enforce network policy without giving any information to the callee who called it.   Those solutions are more like a message broker approach, where you are authenticated to the bus, but the receiver of the message doesn't necessarily have high assurance of who sent the message.   I am particularly trying to call out authentication as "I know who sent this message".